src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/DynamicTable.cs

```diff
@@ -89,11 +90,27 @@ public void Resize(int maxSize)
                 _buffer = newBuffer;
                 _removeIndex = 0;
                 _insertIndex = _count;
-                _maxSize = maxSize;
             }
-            else
+
+            var entry = new HeaderField(staticTableIndex, name, value);
+            _buffer[_insertIndex] = entry;
+
+            if (++_insertIndex == _buffer.Length)
```

Comment:
No feedback, just I'm curious why change the modulo to this? Is there some advantage to have it like this?

Reply:
https://github.com/dotnet/runtime/pull/112719#discussion_r1963835423

It's a microoptimization since modulo is (relatively) that expensive that a predicted branch can be cheaper.
We use a similar trick in `Queue` for example
https://github.com/dotnet/runtime/blob/f0aa3abca0e52afdae7b2e804529529f9d3aaee9/src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Queue.cs#L333-L345